packages/firestore/src/core/sync_engine.ts

```diff
@@ -759,6 +783,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
 
   private async emitNewSnapsAndNotifyLocalStore(
     changes: MaybeDocumentMap,
+    snapshotVersion: SnapshotVersion,
```

Comment:
What is this snapshot version? A read version? A commit version? The highest of either? What happens if a caller passes SnapshotVersion.MIN?

Looking at callers I can't tell if they're passing the right values.

Assuming that passing the commit version and the read version is legit here, should we name this highestVersion? highWatermarkVersion?

Basically, I think naming these new variables and parameters "snapshotVersion" isn't helping: these should always be named after the concrete kind of version they are.

Reply:
It's the version that the backend associates with the given operation - either the global snapshot version or the commit version of the write. I am not sure if the name can be much more specific :/

I did however go back and changed how this method gets its argument. I am using something similar to what I do in multi-tab (applying a "fake" current change via a synthesized RemoteEvent) and have added a helper that creates a "remote event" for an acknowledged write. This remote event is then used to propagate the commit version.
This gets rid of the argument and of some of the ambiguity. What do you think?